### ISSUE TITLE
"Apply Permissions to Enclosed Files" button still does not work

### ISSUE THREAD

**Issue by chronologicaldot** [linuxmint/nemo#2851]
Mint 20.1
64-bit

This time, we're working with a very recently downloaded version of Mint.
Button in Nemo still doesn't work. This problem was reported here but prematurely closed:
https://github.com/linuxmint/nemo/issues/2460
You can look at that page for the steps to reproduce the problem.

If the button isn't going to do anything, it should be removed.

**Comment by LinuxOnTheDesktop** [comment]
To quote from a post above:

> Button in Nemo still doesn't work. [. . . .] You can look at that page for the steps to reproduce the problem.
> If the button isn't going to do anything, it should be removed.

Is this not a rather serious problem?

**Comment by Drugwash2** [comment]
Ah, so finally got the proof that it's not me the stupid that doesn't know which pixel in that button to click precisely in order to get the desired result. _(two trucks of irony implied)_

Commenting on the suggestion made by ©chronologicaldot above I'd say the button label should rather be a generic **Apply permissions recursively** and should work **for both folders and files** according to the settings on the current Properties page, which is what every sane user would assume. I've been banging my head against the wall way too many times wondering why it doesn't work and whether I was too stupid to understand all this inept Linux permissions thing - coming from about 20 years of Windows 98 SE.

Guess I'd have to compile my own (fixed) versions of about everything here considering there's no backporting to Mint 19.2 anyway. But I may not live that long.

**Comment by Jeremy7701** [comment]
Nowhere does it say anything about **recursive** permissions.
Actually, reading the 'man' page about `chmod` isn't very clear about the effect of recursive permissions on directories/files.
...
The letters rwxX select file mode bits for the affected users: read (r), write (w), execute (or  search  for directories)  (x),  execute/search  only if the file is a directory or already has execute permission for some user (X)
????
...

Here is a picture I discovered of the effects of permission bits:-
![permissions](https://user-images.githubusercontent.com/52585119/168591328-2ef7751d-2f0c-49ef-9d32-027d41b58b28.png)
Note that if a directory doesn't have execute rights, then file navigation is essentially impossible.

**Comment by Drugwash2** [comment]
> Nowhere does it say anything about **recursive** permissions.

Yeah, true. Maybe I was expecting too much from a flawed application. Will check out Double Commander in this regard, although from what I've seen it doesn't even display current permissions in the Properties dialog.

**Comment by LinuxOnTheDesktop** [comment]
> Nowhere does it say anything about **recursive** permissions.

Well,

> calls nemo_file_set_permissions_recursive()

So there seem to be at least two problems.

1. The GUI is unclear about what the 'apply to files' options does.
2. The program does not do what its source code suggests that it is designed to do.

That Linux permissions are complex makes _1_ - and indeed _2_ - especially unwelcome.

**Comment by Drugwash2** [comment]
Ah so, again, it seems I wasn't so crazy after all... Intuition was right on spot: it **should** apply recursively.
As mentioned by @chronological dot above, there is a function called `apply_recursive_clicked` as callback for that button's functionality so at least that name would suggest a recursive operation.

You are indeed right about the two points although in my mind button's label always suggested without a doubt a recursive operation.

Now all this aside: what is to do? Obviously there won't be any backports to Mint 19.x so for me it's strictly an academic discussion, but just for curiosity I'd like to see whether anyone cares to fix the - sometimes big, even huge - problems before implementing more and more useless gimmicks. Not referring to Nemo only but to Mint generally and even to Linux as a whole.
